torch/csrc/jit/passes/quantization.cpp

```diff
@@ -128,16 +143,35 @@ static void addObserverFor(Value* v, Node* original_observer_node) {
 }
 
 static bool outputsNeedToBeObserved(Node* n) {
-  return n->kind() != prim::Constant;
+  return n->kind() != prim::Constant && n->kind() != prim::PythonOp;
 }
 
-void InsertObserverNodes(std::shared_ptr<Graph>& graph, Node* observer_node) {
+void InsertObserverNodes(const std::unique_ptr<script::Method>& method,
+  Node* observer_node) {
+  AT_ASSERT(method != nullptr);
+  auto graph = method->graph();
+  AT_ASSERT(graph != nullptr);
   // For storing all values that need to be instrumented with an observer call.
   std::vector<Value*> values_to_observe;
 
   // For traversing all blocks in the graph including subblocks.
   std::stack<Block*> blocks_to_visit;
 
+  // Add observer for external input nodes excluding parameters
+  // These are treated as activation as they vary across batches
+  // and need to be observed.
+
+  // prim::Param nodes do not belong to the graph. Hence the Insert
+  // point is the beginning of graph node. This also safe guards against
+  // observing a potentially mutated value due to some in-place operation
+  Node* insert_node = *graph->nodes().begin();
+  for (auto idx = 0; idx < method->num_inputs(); ++idx) {
```

Comment:
Please don't use `auto` in such loops. It is resolved to `signed int` here, while the comparison is against `unsigned int64_t`. While it probably won't cause problems in this particular case, it is a better practice to explicitly specify type in such cases.